Parse/Internal/Object/Subclassing/PFObjectSubclassingController.m

```diff
@@ -122,6 +113,33 @@ + (void)clearDefaultController {
     return result;
 }
 
+- (void)scanForUnregisteredSubclasses:(BOOL)shouldSubscribe {
+    // NOTE: Potential race-condition here - if another thread dynamically loads a bundle, we may end up accidentally
+    // Skipping a bundle. Not entirely sure of the best solution to that here.
+    if (shouldSubscribe) {
+        @weakify(self);
+        [[NSNotificationCenter defaultCenter] addObserverForName:NSBundleDidLoadNotification
```

Comment:
From offline discussion:
Store the registration token in the ivar, validate that it's `nil` here to avoid double subscribe as well as clean up the registration in `dealloc`.